app/lib/server/functions/notifications/email.js

```diff
@@ -55,13 +59,21 @@ function getEmailContent({ message, user, room }) {
 			lng,
 		});
 
+		if (!settings.get('Email_notification_show_message')) {
+			return fileHeader;
+		}
+
 		let content = `${ s.escapeHTML(message.file.name) }`;
 
 		if (message.attachments && message.attachments.length === 1 && message.attachments[0].description !== '') {
 			content += `<br/><br/>${ s.escapeHTML(message.attachments[0].description) }`;
 		}
 
-		return `${ fileHeader }<br/><br/>${ content }`;
+		return `${ fileHeader }:<br/><br/>${ content }`;
+	}
+
+	if (!settings.get('Email_notification_show_message')) {
```

Comment:
I think moving this `if` to above [if (message.msg !== '') {](https://github.com/RocketChat/Rocket.Chat/pull/15406/files#diff-b09b1228639562af1fb18b68db6cd363R34) is the only check of the new setting that needs to be done, what do you think?

Reply:
The middle `if` returns a different header from the other two